File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.rb
# @param {Integer[]} gifts
# @param {Integer} k
# @return {Integer}
def pick_gifts(gifts, k)
    while k > 0
    max_pile_index = gifts.index(gifts.max)
    gifts[max_pile_index] = Math.sqrt(gifts[max_pile_index]).floor
    k-=1
    end 
    gifts.sum
end

File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        while (k > 0) {
            auto max_pile = max_element(gifts.begin(), gifts.end());
            int max_pile_index = distance(gifts.begin(), max_pile);
            gifts[max_pile_index] = floor(sqrt(gifts[max_pile_index]));
            k--;
        }
        long long sum = accumulate(gifts.begin(), gifts.end(), 0LL);
        return sum;
    }
};

 